File: emirhan-kudun-portfolio/sources/polyglot/php/source_contract.php
<?php

declare(strict_types=1);

function seis_source_contract(string $id, string $pluginRef, string $layer): array
{
    return [
        'id' => $id,
        'pluginRef' => $pluginRef,
        'layer' => $layer,
        'visibleInPortfolio' => str_starts_with($pluginRef, 'plugin://'),
    ];
}

File: emirhan-kudun-portfolio/sources/polyglot/kotlin/SourceContract.kt
package seis.portfolio.sources

data class SourceContract(
    val id: String,
    val pluginRef: String,
    val layer: String,
    val visibleInPortfolio: Boolean
) {
    val hasValidPluginRef: Boolean
        get() = pluginRef.startsWith("plugin://")
}
